src/analysis/retail/warlock/demonology/modules/guide/Cooldowns.tsx

```diff
@@ -0,0 +1,59 @@
+import { Section, SubSection, useAnalyzer, useInfo } from 'interface/guide';
+import CastEfficiency from 'parser/shared/modules/CastEfficiency';
+import TALENTS from 'common/TALENTS/warlock';
+import { Talent } from 'common/TALENTS/types';
+import CastEfficiencyBar from 'parser/ui/CastEfficiencyBar';
+import { GapHighlight } from 'parser/ui/CooldownBar';
+
+interface Props {
+  checklist: Talent[];
+}
+
+const COOLDOWNS: Talent[] = [
+  TALENTS.NETHER_PORTAL_TALENT,
+  TALENTS.GRIMOIRE_FELGUARD_TALENT,
+  TALENTS.SUMMON_VILEFIEND_TALENT,
+  TALENTS.SUMMON_DEMONIC_TYRANT_TALENT,
+  TALENTS.DEMONIC_STRENGTH_TALENT,
+  TALENTS.POWER_SIPHON_TALENT,
+  TALENTS.GUILLOTINE_TALENT,
+];
+
+function Cooldowns() {
+  return (
+    <Section title="Cooldowns">
+      <p>
+        <strong>Cooldowns</strong> - this graph shows when you used your major cooldowns and how
+        long you waited to use them again. Unless you're holding these for specific raid events, try
+        to use these on as soon as they become available.
+      </p>
+      <CooldownGraphSubsection checklist={COOLDOWNS} />
+    </Section>
+  );
+}
+
+const CooldownGraphSubsection = ({ checklist }: Props) => {
```

Comment:
You can simplify this a bit by using the `CooldownGraphSubsection` from core now. You can do something like this:

```tsx
import { TALENTS_DEMON_HUNTER } from 'common/TALENTS/demonhunter';
import CoreCooldownGraphSubsection, {
  Cooldown,
} from 'interface/guide/components/CooldownGraphSubSection';

const cooldowns: Cooldown[] = [
  {
    spell: TALENTS_DEMON_HUNTER.SOUL_CARVER_TALENT,
    isActive: (c) => c.hasTalent(TALENTS_DEMON_HUNTER.SOUL_CARVER_TALENT),
  },
  {
    spell: TALENTS_DEMON_HUNTER.FEL_DEVASTATION_TALENT,
    isActive: (c) => c.hasTalent(TALENTS_DEMON_HUNTER.FEL_DEVASTATION_TALENT),
  },
  {
    spell: TALENTS_DEMON_HUNTER.ELYSIAN_DECREE_TALENT,
    isActive: (c) => c.hasTalent(TALENTS_DEMON_HUNTER.ELYSIAN_DECREE_TALENT),
  },
  {
    spell: TALENTS_DEMON_HUNTER.THE_HUNT_TALENT,
    isActive: (c) => c.hasTalent(TALENTS_DEMON_HUNTER.THE_HUNT_TALENT),
  },
  {
    spell: TALENTS_DEMON_HUNTER.SOUL_BARRIER_TALENT,
    isActive: (c) => c.hasTalent(TALENTS_DEMON_HUNTER.SOUL_BARRIER_TALENT),
  },
  {
    spell: TALENTS_DEMON_HUNTER.BULK_EXTRACTION_TALENT,
    isActive: (c) => c.hasTalent(TALENTS_DEMON_HUNTER.BULK_EXTRACTION_TALENT),
  },
  {
    spell: TALENTS_DEMON_HUNTER.FIERY_BRAND_TALENT,
    isActive: (c) =>
      c.hasTalent(TALENTS_DEMON_HUNTER.FIERY_BRAND_TALENT) &&
      c.hasTalent(TALENTS_DEMON_HUNTER.FIERY_DEMISE_TALENT),
  },
];

const CooldownGraphSubsection = () => <CoreCooldownGraphSubsection cooldowns={cooldowns} />;
```

Reply:
Oh I'll have a look!